packages/core/src/anchor/ethereum/ethereum-anchor-service.ts

```diff
@@ -75,6 +78,23 @@ export default class EthereumAnchorService extends AnchorService {
         this._ceramic = ceramic
     }
 
+    async init(): Promise<void> {
+        // Get the chainIds supported by our anchor service
+        const response = await fetch(this.chainIdApiEndpoint)
+        const json = await response.json()
+        this._supportedChains = json.supportedChains
+
+        // Confirm that we have an eth provider that works for each of the chains that the anchor service supports
+        for (const chain of this._supportedChains) {
+            const provider = this._getEthProvider(chain)
+            const provider_chain_idnum = (await provider.getNetwork()).chainId
+            const provider_chain = BASE_CHAIN_ID + ':' + provider_chain_idnum
+            if (chain != provider_chain) {
```

Comment:
If one ethereum rpc is passed (chainId 1), and CAS returns multiple supported chains (chainId 1, 2, 3), this would always fail on [2, 3].

Reply:
Guess, this thing with supported chains and checking of support from providers is totally unnecessary. Have ethereumRpc => use this provider (in `_getEthereumProvider`), check if it supports the chainId of validated anchor. If supports, use it. If it does not, use fallback provider.

Reply:
Note for fixing this: You can call `eth_chainId` to get the chainId of the provider: https://github.com/ceramicnetwork/js-ceramic/blob/develop/packages/blockchain-utils-linking/src/ethereum.ts#L31

Reply:
I made it explicit that we only support one chain at a time right now

Reply:
In the next two weeks, the comment in this pull request is forgotten by everyone, but code remains. It would be fun to watch a person doing digital archeology to figure the reasoning here.

Reply:
If one chain, the loop does not make sense. If multiple chains, it conveys a misleading semantics.

Reply:
the relation implied here is `some` or `includes`, not `every` which is currently encoded.

Reply:
> the relation implied here is `some` or `includes`, not `every` which is currently encoded.

It actually should be `every`.  If the CAS supported multiple chains, then ceramic would need to be prepared to validate any anchorProofs from any of the supported chains that might get used.

Reply:
I'll make a change to get the first element of the array returned from the CAS and from that point on structure the code to assume only one supported chain, so the code more explicitly matches the current state of the world, rather than trying to code for a hypothetical future we don't yet have

Reply:
> If the CAS supported multiple chains, then ceramic would need to be prepared to validate any anchorProofs from any of the supported chains that might get used.

Mea culpa. That's right. I was stuck on current "return provider just based on single ethereum rpc param" logic in `getEthereumProvider`.